### PR TITLE
MULE-18875: ValueProvider/SampleData optional parameters declared in resolver are not included in the model

### PR DESCRIPTION
**Pull request by gabrieldalborgo**

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ValueProvidersParameterDeclarationEnricher.java [hunk 1]
```diff
@@ -155,9 +162,9 @@ private void enrichParameter(OfValues resolverClass,
     paramDeclaration.addModelProperty(propertyBuilder.build());
 
     valueProviderModelConsumer
-        .accept(new ValueProviderModel(getRequiredParametersAliases(resolverParameters, containerParameterNames),
```
Comment (ndinu) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ValueProvidersParameterDeclarationEnricher.java:158]: I think we should add tests that assert the value provider model other than the check of the test extensions extension model.

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ValueProvidersParameterDeclarationEnricher.java [hunk 2]
```diff
@@ -247,11 +254,17 @@ private void enrichParameter(OfValues resolverClass,
...
-        .filter(ExtensionParameter::isRequired)
-        .map(param -> parameterNames.getOrDefault(param.getName(), param.getName()))
+  private List<ParameterModel> getParametersModel(List<ExtensionParameter> parameterDeclarations,
+                                                  Map<String, String> parameterNames,
+                                                  List<ParameterDeclaration> allParameters) {
+    Map<String, Boolean> paramsInfo = parameterDeclarations.stream()
+        .collect(toMap(param -> parameterNames.getOrDefault(param.getName(), param.getName()), ExtensionParameter::isRequired));
+    return allParameters.stream()
+        .filter(param -> paramsInfo.containsKey(param.getName()))
+        .map(param -> new ImmutableParameterModel(param.getName(), param.getDescription(), param.getType(),
```
[the remark this answers is not in the document]
Comment (gabrieldalborgo) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ValueProvidersParameterDeclarationEnricher.java:264]: What do you have in mind? Inherit the default value from the real parameter? or allow to define a new default value at the ValueProvider?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ValueProvidersParameterDeclarationEnricher.java [hunk 3]
```diff
@@ -247,11 +254,17 @@ private void enrichParameter(OfValues resolverClass,
...
+                                                  param.hasDynamicType(), paramsInfo.get(param.getName()),
```
[the remark this answers is not in the document]
Comment (gabrieldalborgo) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ValueProvidersParameterDeclarationEnricher.java:265]: Yes, that is right

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ValueProvidersParameterDeclarationEnricher.java (hunk 1 above)
Comment (gabrieldalborgo) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ValueProvidersParameterDeclarationEnricher.java:158]: Where would you create those tests?

Comment (ndinu) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ValueProvidersParameterDeclarationEnricher.java:158]: I would create a test for the enricher where the extension declaration is mocked, and then enriched, then assert the enriched declaration.

Comment (gabrieldalborgo) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ValueProvidersParameterDeclarationEnricher.java:158]: Tests added

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ValueProvidersParameterDeclarationEnricher.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ndinu) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ValueProvidersParameterDeclarationEnricher.java:264]: I had in mind setting the default value of the real parameter, which is the value that will be taken for values resolution is none is provided.

File: modules/extensions-support/pom.xml [hunk 4]
```diff
@@ -211,6 +218,24 @@
...
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.mule.tests</groupId>
+            <artifactId>mule-values-extension</artifactId>
+            <version>4.4.0-SNAPSHOT</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.mule.tests</groupId>
```
Comment (marianogonzalez) [inline modules/extensions-support/pom.xml:228]: repeated dependency

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/SampleDataDeclarationEnricher.java [hunk 5]
```diff
@@ -156,11 +161,17 @@ private SampleDataProviderModel createSampleDataModel(SampleData annotation, Exe
...
-        .filter(ExtensionParameter::isRequired)
-        .map(param -> parameterNames.getOrDefault(param.getName(), param.getName()))
+  private List<ParameterModel> getParametersModel(List<ExtensionParameter> parameterDeclarations,
+                                                  Map<String, String> parameterNames,
+                                                  List<ParameterDeclaration> allParameters) {
+    Map<String, Boolean> paramsInfo = parameterDeclarations.stream()
+        .collect(toMap(param -> parameterNames.getOrDefault(param.getName(), param.getName()), ExtensionParameter::isRequired));
+    return allParameters.stream()
+        .filter(param -> paramsInfo.containsKey(param.getName()))
+        .map(param -> new ImmutableParameterModel(param.getName(), param.getDescription(), param.getType(),
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/SampleDataDeclarationEnricher.java:171]: you should use the aliased name here instead of `param.getName()`.

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ValueProvidersParameterDeclarationEnricher.java [hunk 6]
```diff
@@ -247,11 +254,19 @@ private void enrichParameter(OfValues resolverClass,
...
-  private List<String> getRequiredParametersAliases(List<ExtensionParameter> parameterDeclarations,
-                                                    Map<String, String> parameterNames) {
-    return parameterDeclarations.stream()
-        .filter(ExtensionParameter::isRequired)
-        .map(param -> parameterNames.getOrDefault(param.getName(), param.getName()))
+  private List<ParameterModel> getParametersModel(List<ExtensionParameter> parameterDeclarations,
+                                                  Map<String, String> parameterNames,
+                                                  List<ParameterDeclaration> allParameters) {
+    Map<String, Boolean> paramsInfo = parameterDeclarations.stream()
+        .collect(toMap(param -> parameterNames.getOrDefault(param.getName(), param.getName()), ExtensionParameter::isRequired));
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ValueProvidersParameterDeclarationEnricher.java:261]: same as before. You're loosing the aliased name. Several test cases should have broken because of this. If not, we have a coverage problem you should fix as part of this PR

**Review by marianogonzalez — CHANGES_REQUESTED** [review]

File: modules/extensions-support/pom.xml [hunk 7]
```diff
@@ -169,6 +169,13 @@
             <classifier>mule-plugin</classifier>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.mule.tests</groupId>
+            <artifactId>mule-values-extension</artifactId>
+            <version>4.4.0-SNAPSHOT</version>
+            <classifier>mule-plugin</classifier>
```
Comment (marianogonzalez) [inline modules/extensions-support/pom.xml:176]: why do you need both the mule-plugin and jar dependencies?

**Review by marianogonzalez — CHANGES_REQUESTED** [review]

**Review by marianogonzalez — APPROVED** [review]

**Comment by gabrieldalborgo** [comment]
--validate

File: modules/extensions-spring-support/src/test/resources/models/metadata.json [hunk 8]
```diff
@@ -38647,7 +38647,6 @@
       "format": "xml",
       "type": "Object",
       "annotations": {
-        "typeId": "#root:shiporder",
```
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/test/resources/models/metadata.json:38650]: Why was this moved?

**Review by marianogonzalez — CHANGES_REQUESTED** [review]

File: modules/extensions-spring-support/src/test/resources/models/metadata.json (hunk 8 above)
Comment (gabrieldalborgo) [reply modules/extensions-spring-support/src/test/resources/models/metadata.json]: I'm not sure, this happens when the extension model is regenerated. It is something that is already in the master. I will create a ticket to review it.
Also, I reverted the changes to this file.

**Review by marianogonzalez — APPROVED** [review]
